### ISSUE TITLE
How to access an environment variable from a filter plugin?

### ISSUE THREAD

**Issue by progre55** [ansible/ansible#2026]
I used to get HOME using $ENV(HOME) and it used to work fine before version 1.0. However, now it returns an array, but I cannot even get the 0's element out of it. Not sure if it matters, but I'm on a Mac OS X.

Here is a sample play:

``` YAML
  vars:
   - user_home: $ENV(HOME)
   - project_home: ${user_home}/project
   - project1_home: $user_home.0/project
   - project2_home: ${user_home.0}/project
   - project3_home: ${user_home[0]}/project
   - project4_home: ${user_home}[0]/project

  tasks:

  - name: Echo homes
    command: /usr/bin/env echo  project_home is ${project_home} project1_home is ${project1_home} project2_home is ${project2_home} project3_home is ${project3_home} project4_home is ${project4_home}
```

and here is the output:

``` bash
"stdout": "project_home is [/Users/progre55]/project project1_home is [/Users/progre55].0/project project2_home is ${user_home.0}/project project3_home is $/project project4_home is [/Users/progre55][0]/project"
```

**Comment by mpdehaan** [comment]
see comments on #2027, fix pending most likely tonight.

You can use "with_env" until then.

(but I agree that's not as good)

**Comment by mpdehaan** [comment]
Fixed in e6ead2c
